### PR TITLE
Various attribute handling adjustments

### PR DESCRIPTION
**Pull request by evs-chris**
Fat fingered the first submission. Meh.

### CODE REVIEW

**Comment by evs-chris** [comment]
This should address #1733 and #1727. Attribute handling is a little bit fiddly, but I think this balances everything fairly well.

* Empty string attributes now keep the empty string instead of becoming 0. That should cover #1733
* Valueless attributes still come out as 0, but components treat that as true now.
* contenteditable elements with no value binding don't get a bonus `undefined` in their `toHTML` output (stumbled across while assembling a test)

Does that sound reasonable?

Sorry about the noise in the test files. I delinted them.

**Comment by Rich-Harris** [comment]
Yep, this looks great. Ready to merge?

File: test/__tests/components.js [hunk 1]
```diff
@@ -830,3 +829,23 @@ test( 'Component mappings used in computations resolve correctly with the mappin
 
 	t.htmlEqual( fixture.innerHTML, ractive.toHTML() );
 });
+
+test( 'Component attributes with no = are boolean true', t => {
+	new Ractive({
+		el: fixture,
+		template: '<cmp foo />',
+		components: { cmp: Ractive.extend({ template: '{{#foo}}yep{{/}}' }) }
```
Comment (martypdx) [inline test/__tests/components.js]: How about `{{# foo === true }}` instead?

**Comment by evs-chris** [comment]
Good to go.

File: test/__tests/components.js (hunk 1 above)
Comment (Rich-Harris) [inline test/__tests/components.js]: yeah that makes sense - I updated it

**Pull request merged**